subprojects/antlr/src/main/java/org/gradle/api/plugins/antlr/AntlrPlugin.java

```diff
@@ -48,37 +47,21 @@ public AntlrPlugin(ObjectFactory objectFactory) {
 
     @Override
     public void apply(final Project project) {
-        project.getPluginManager().apply(JavaPlugin.class);
+        project.getPluginManager().apply(JavaLibraryPlugin.class);
 
         // set up a configuration named 'antlr' for the user to specify the antlr libs to use in case
         // they want a specific version etc.
         final Configuration antlrConfiguration = project.getConfigurations().create(ANTLR_CONFIGURATION_NAME)
                 .setVisible(false)
                 .setDescription("The Antlr libraries to be used for this project.");
 
-        antlrConfiguration.defaultDependencies(new Action<DependencySet>() {
-            @Override
-            public void execute(DependencySet dependencies) {
-                dependencies.add(project.getDependencies().create("antlr:antlr:2.7.7@jar"));
-            }
-        });
+        antlrConfiguration.defaultDependencies(dependencies -> dependencies.add(project.getDependencies().create("antlr:antlr:2.7.7@jar")));
 
-        @SuppressWarnings("deprecation")
-        Configuration compileConfiguration = project.getConfigurations().getByName(JavaPlugin.COMPILE_CONFIGURATION_NAME);
-        compileConfiguration.extendsFrom(antlrConfiguration);
+        Configuration apiConfiguration = project.getConfigurations().getByName(JavaPlugin.API_CONFIGURATION_NAME);
```

Comment:
So using Antlr leaks to consumer.
Is this something we should deprecate as well?

Reply:
Antlr dependencies leak to consumers but they are effectively part of the API.

Reply:
I agree that this test is useless. It does not do what its name says.